astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -640,3 +642,13 @@ def test_assignment_with_different_names(self):
         assert qra[1].value == np.array((-1000.0, -2000.0, -3000.0), qra.dtype)
         # Ensure we do not override dtype names of value.
         assert value.dtype.names == ("x", "y", "z")
+
+
+@pytest.mark.filterwarnings("ignore::UserWarning")
+@pytest.mark.skipif(not HAS_ARRAY_API_COMPAT, reason="requires array-api-compat")
+def test_array_api_init():
+    import numpy.array_api as np
```

Comment:
Maybe call it different to avoid confusion. Or `from numpy.array_api import asarray`